src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!this.isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
Pedantic approach would be to iterate over `BusyItems` first and `enforce`, and after that call a delegate from within `assert` that iterates over `IdleItems` only. Latter indeed is not possible if rest of the scheduler is in sane state but can serve as an additional debugging helper (== ``assert`).